mmeval/metrics/instance_seg.py

```diff
@@ -0,0 +1,288 @@
+        ...         ann_info_data = dict()
+        ...         ann_info_data['pts_instance_mask'] = torch.tensor(
+        ...             gt_instance_mask)
+        ...         ann_info_data['pts_semantic_mask'] = torch.tensor(
+        ...             gt_semantic_mask)
+        ...
+        ...         results_dict = dict()
+        ...         pred_instance_mask = np.ones(n_points, dtype=int) * -1
+        ...         labels = []
+        ...         scores = []
+        ...         for i, gt_label in enumerate(gt_labels):
+        ...             begin = i * 300
+        ...             end = begin + 300
+        ...             pred_instance_mask[begin:end] = i
+        ...             labels.append(gt_label)
+        ...             scores.append(.99)
+        ...
+        ...         results_dict['pts_instance_mask'] = torch.tensor(
+        ...             pred_instance_mask)
+        ...         results_dict['instance_labels'] = torch.tensor(labels)
+        ...         results_dict['instance_scores'] = torch.tensor(scores)
+        ...
+        ...         predictions.append(results_dict)
+        ...         groundtruths.append(ann_info_data)
+        ...
+        ...     return predictions, groundtruths
+        >>>
+        >>> instance_seg_metric = InstanceSegMetric(dataset_meta=dataset_meta)
+        >>> res = instance_seg_metric(predictions, groundtruths)
+        >>> res
+        {
+        'all_ap': 1.0,
+        'all_ap_50%': 1.0,
+        'all_ap_25%': 1.0,
+        'classes': {
+            'cabinet': {
+                'ap': 1.0,
+                'ap50%': 1.0,
+                'ap25%': 1.0
+            },
+            'bed': {
+                'ap': 1.0,
+                'ap50%': 1.0,
+                'ap25%': 1.0
+            },
+            'chair': {
+                'ap': 1.0,
+                'ap50%': 1.0,
+                'ap25%': 1.0
+            },
+            ...
+        }
+    }
+    """
+
+    def __init__(self,
+                 classes: Optional[List[str]] = None,
+                 valid_class_ids: Optional[List[int]] = None,
+                 **kwargs):
+        super().__init__(**kwargs)
+        assert (self.dataset_meta
+                is not None) or (classes is not None
+                                 and valid_class_ids is not None)
+        if self.dataset_meta is not None:
+            if classes is not None or valid_class_ids is not None:
+                warnings.warn('`classes` and `valid_class_ids` are ignored.')
+            self.classes = self.dataset_meta['classes']
+            self.valid_class_ids = self.dataset_meta['seg_valid_class_ids']
+        else:
+            if not isinstance(classes, list):
+                raise TypeError('`classes` should be List[str]')
+            if not isinstance(valid_class_ids, list):
+                raise TypeError('`valid_class_ids` should be List[int]')
+            self.classes = classes
+            self.valid_class_ids = valid_class_ids
```

Comment:
We can set a classes property to address this, please ref: https://github.com/open-mmlab/mmeval/blob/e635ba6338105d92f053b46320f3b1ca340ef9b4/mmeval/metrics/mean_iou.py#L114-L138

Reply:
fixed